integration-tests/src/test_loop/utils/trie_sanity.rs

```diff
@@ -340,6 +343,8 @@ fn should_assert_state_sanity(
 /// Asserts that all parent shard State is accessible via parent and children shards.
 pub fn check_state_shard_uid_mapping_after_resharding(
     client: &Client,
+    prev_block_hash: &CryptoHash,
+    resharding_block_hash: &CryptoHash,
```

Comment:
It feels strange to pass the prev_block_hash and resharding_block_hash, and get the tip from the client on the first line of this method. Mixing "pure"-ish function approach with "stateful" approach is asking for trouble.

Reply:
fixed in fa8e29e23c81